dbms/src/Storages/MergeTree/MergeTreePartsMover.cpp

```diff
@@ -94,35 +94,91 @@ bool MergeTreePartsMover::selectPartsForMove(
     {
         for (const auto & disk : volumes[i]->disks)
         {
-            UInt64 required_available_space = disk->getTotalSpace() * policy->getMoveFactor();
+            UInt64 required_maximum_available_space = disk->getTotalSpace() * policy->getMoveFactor();
             UInt64 unreserved_space = disk->getUnreservedSpace();
 
-            if (required_available_space > unreserved_space)
-                need_to_move.emplace(disk, required_available_space - unreserved_space);
+            if (unreserved_space < required_maximum_available_space)
+                need_to_move.emplace(disk, required_maximum_available_space - unreserved_space);
         }
     }
 
+    auto current_time = time(nullptr);
+
     for (const auto & part : data_parts)
     {
         String reason;
         /// Don't report message to log, because logging is excessive
         if (!can_move(part, &reason))
             continue;
 
+        const auto ttl_entries_end = part->storage.move_ttl_entries_by_name.end();
+        auto best_ttl_entry_it = ttl_entries_end;
+        time_t max_max_ttl = 0;
```

Comment:
???

Reply:
That was a mistake.

Reply:
I moved this code here:
https://github.com/excitoon-favorites/ClickHouse/blob/d1bf3c00950ba02f43769eec63fcc37c2c9ef133/dbms/src/Storages/MergeTree/MergeTreeData.cpp#L3678-L3699